src/openrct2-ui/windows/MapGen.cpp

```diff
@@ -818,7 +817,7 @@ static void window_mapgen_simplex_mouseup(rct_window* w, rct_widgetindex widgetI
             mapgenSettings.mapSize = _mapSize;
 
             mapgenSettings.height = _baseHeight;
-            mapgenSettings.water_level = _waterLevel + 2;
+            mapgenSettings.water_level = _waterLevel + MINIMUM_WATER_HEIGHT;
```

Comment:
The changes in this file (and similar ones) should be backported.

Reply:
This was introduces in https://github.com/OpenRCT2/OpenRCT2/pull/14700, when raising the water height limit. This commit can safely be cherry-picked: 6a64b5a63cf24d7a9baf8f398445d341fc65c179

Reply:
OK, can you explain the usage `MINIMUM_WATER_HEIGHT + MAXIMUM_WATER_HEIGHT` to me? It’s in a place where I would expect it to just read `MAXIMUM_WATER_HEIGHT`.

Reply:
Good question, I was thinking the same thing yesterday. The game originally stores the water level in 5 bits, each step increases the base height by 2, and 0 means there is no water at all. This leaves us with 31 usable water levels. Since in the original games, surface element cannot have a base height under 2, a minimum water level of the same base height (value 1) would be useless. Instead of wasting this value, Chris Sawyer decided to add a minimum water level which essentially is an offset. With this offset, we gain one additional usable water level while still only using 5 bits.

The naming of `MAXIMUM_WATER_HEIGHT` is a bit confusing, as it's not the absolute height, but the maximum offset from the minimum water level.

Reply:
| Value | Water surface height |
| --- | ---
| 0 | None |
| 1 | 4 |
| 2 | 6 |
| 3 | 8 |
| ... | ... |
| 29 | 60 |
| 30 | 62 |
| 31 | 64 |

Reply:
I see. That sounds like a good candidate for refactoring - the offset is a detail that should only be known to the surface tile element itself. I might create an issue for that.